src/components/Map.tsx

```diff
@@ -178,20 +204,28 @@ const MapContainer = (props) => {
     }
     drawOnMap(roadOk, roadWarn, roadSnow);
   };
-
-  const drawOnMap = (roadOk, roadWarn, roadSnow) => {
-    map.current.getSource("snow-plow-ok").setData({
-      type: "FeatureCollection",
-      features: roadOk,
-    });
-    map.current.getSource("snow-plow-warn").setData({
-      type: "FeatureCollection",
-      features: roadWarn,
-    });
-    map.current.getSource("snow-plow-snow").setData({
-      type: "FeatureCollection",
-      features: roadSnow,
-    });
+  const drawOnMap = (
+    roadOk: SimulationFeature[],
+    roadWarn: SimulationFeature[],
+    roadSnow: SimulationFeature[]
+  ) => {
+    if (map.current !== null) {
+      // @ts-expect-error TODO: Are we sure setData works here?
```

Comment:
I found something: https://github.com/maplibre/maplibre-gl-js/issues/785#issuecomment-1066511246

Reply:
Excellent! I think that would work! You wanna have a go at it? 😊